src/components/data/SelectionDrawer.js

```diff
@@ -41,6 +40,64 @@ const useStyles = makeStyles(styles);
 
 const PAGINATION_BAR_HEIGHT = 60;
 
+function getInvalidSelectionCount(
+    getSelectedResources,
+    acceptedType,
+    acceptedInfoTypes,
+    multiSelect
+) {
+    const selectedResources = getSelectedResources();
+    const allowSelectionByInfoTypes =
+        acceptedInfoTypes && acceptedInfoTypes?.length > 0;
+    const invalidInfoTypeSelections = allowSelectionByInfoTypes
+        ? selectedResources.filter(
+              (resource) => !acceptedInfoTypes?.includes(resource?.infoType)
+          ).length
+        : 0;
+    let invalidTotal = 0;
+
+    switch (acceptedType) {
+        case ResourceTypes.FILE:
+            //for file inputs, we want all the selections to be of 'file' type
+            // and in one of the accepted info types if specified
+            invalidTotal =
+                selectedResources.filter(
+                    (resource) => resource.type.toLowerCase() !== acceptedType
+                ).length || invalidInfoTypeSelections;
+            break;
+        case ResourceTypes.FOLDER:
+            if (!multiSelect) {
+                const selectedType = selectedResources[0]?.type;
+                //for single folder input selection for Apps,
+                //we allow either HT-Path-list file or a single folder as input
+                if (allowSelectionByInfoTypes) {
+                    if (
+                        invalidInfoTypeSelections > 0 &&
+                        selectedType !== ResourceTypes.FOLDER
+                    ) {
+                        invalidTotal = 1;
+                    }
+                } else {
+                    // in other places like Move or Preferences default output folder selection
+                    // we only allow folders
+                    if (selectedType !== ResourceTypes.FOLDER) {
+                        invalidTotal = 1;
+                    }
+                }
+            } else {
+                //we don't support info type based selections for multi-select folders
```

Comment:
I think this works, but I also think it makes sense to have the responsibility be on the parent component to not provide a list of acceptable info types if they aren't actually acceptable.  

If that were the case, then I think this `switch` could be replaced with something like:
```
invalidTotal = acceptedType !== ResourceTypes.ANY 
    ? selectedResources.filter((resource) => resource.type.toLowerCase() !== acceptedType && !acceptedInfoTypes.includes(resource.infoType)) 
    : 0;
```

Reply:
This also makes sense to me.

I would also prefer to check each item in the selected resources for both `acceptedType` and `acceptedInfoTypes`, rather than checking a count of invalid InfoTypes separately from accepted Types; otherwise I think there's a risk that the invalid selection count might be off.

Suppose in the future we allow app integrators to restrict file inputs by InfoType. If the user selects both a folder and a file with the wrong InfoType, then I think the logic in the `case ResourceTypes.FILE` will only report 1 invalid selection.

Putting the checks on each item in a single `filter` loop should correct that.

Reply:
Actually we will still need to do a `switch` even with this suggested change:
- for the case described by @psarando, we must check if the selected item is of type `FILE` `and` it conforms to the accepted infoType array if provided.
- for a `FOLDER` input type,  we must check if the selected item is of type `FOLDER` `or` it conforms to the accepted infoType array if provided. 
Beyond this, the filtering logic must still account of `null` or `undefined` `acceptedInfoTypes`.

Let me know if this makes sense.